### DISCUSSION TITLE
Feedforward for arduino TC4+ based  PID

### DISCUSSION THREAD

**Discussion by williamcramer** [artisan-roaster-scope/artisan#1305] [Q&A]
Hi folks,

I have moved to artisan (recently) 2.8.4, connected to an arduino TC4+ driving a Quest M3 via SSR's and an AC dimmer for the fan. I'm finding cool down is a lot faster than warm up, so I was wondering of anyone knows how to configure a feedforward in Artisan or the TC4? Ideally PID should ramp up and down equally, but I'm finding ramp up is way slower than ramp down, so tuning is challenging. Apparently (I am not an expert on either theory or practice) the classic solution in PID is a feed forward loop to push the system closer to equilibrium, but can find no info on it in Artisan discussions.

Has this ever been tried, forked or requested?

**Comment by MAKOMO** [comment]
You could replace the PID on your TC4 with [one that supports feedforward](https://github.com/bmbeyers/PIDwFF). Anyhow, a PID (especially the Artisan built-in PID due to its low loop speed) are maybe not the ideal instrument to control slow drum roasters (especially electric ones). I regard the asymmetry of the process as a minor complication compared to the others.

Anyhow, the project is open for contributions via PRs and at any time you can fork and build your own modified Artisan.
